src/plugins/hetero/src/plugin.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+#include <memory>
+#include <vector>
+#include <map>
+#include <string>
+#include <utility>
+#include <fstream>
+#include <unordered_set>
+
+// LEGACY
+#include "ie_metric_helpers.hpp"
+// LEGACY
+
+#include "openvino/runtime/device_id_parser.hpp"
+#include "openvino/runtime/properties.hpp"
+#include "openvino/runtime/internal_properties.hpp"
+#include "openvino/util/common_util.hpp"
+
+#include "plugin.hpp"
+// #include "compiled_model.hpp"
+#include "itt.hpp"
+// #include "properties.hpp"
+
+// TODO (vurusovs) required for conversion to legacy API 1.0
+#include "converter_utils.hpp"
+#include "plugin.hpp"
+#include "../executable_network.hpp"
+#include "../internal_properties.hpp"
+
+#include "iplugin_wrapper.hpp"
+// TODO (vurusovs) required for conversion to legacy API 1.0
+
+
+ov::hetero::Plugin::Plugin() {
+    set_device_name("HETERO");
+}
+
+std::shared_ptr<ov::ICompiledModel> ov::hetero::Plugin::compile_model(
+    const std::shared_ptr<const ov::Model>& model,
+    const ov::AnyMap& properties) const {
+    OV_ITT_SCOPED_TASK(itt::domains::Hetero, "Plugin::compile_model");
+
+    // auto fullConfig = ov::hetero::Configuration{properties, m_cfg};
+    // auto compiled_model = std::make_shared<CompiledModel>(
+    //     model->clone(),
+    //     shared_from_this(),
+    //     fullConfig);
+    // return compiled_model;
+    
+    auto shared_this = std::const_pointer_cast<ov::IPlugin>(shared_from_this());
```

Comment:
Why do you need const cast in new classes? Right now it can be added on the executable network side and removed in future.